instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/SqsAccess.java

```diff
@@ -8,12 +8,18 @@
 import io.opentelemetry.context.propagation.TextMapPropagator;
 import io.opentelemetry.javaagent.tooling.muzzle.NoMuzzle;
 import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.SdkResponse;
 import software.amazon.awssdk.core.interceptor.Context;
 import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
 
 // helper class for calling methods that use sqs types in SqsImpl
 // if SqsImpl is not present these methods are no op
 final class SqsAccess {
+  private SqsAccess() {}
+
   private static final boolean enabled = isSqsImplPresent();
```

Comment:
This is actually not working reliably. If the first AWS SDK request being executed is not an SQS-one, due to the changes here, we will still load SqsAccess to check whether it is one. This then checks if SqsImpl can be loaded, which then fails because the SQS advice did not apply yet.

There are a few ways to solve this, but I'm not sure what is best, any opinions @laurit? I can think of these:
1. Eagerly load the SQS advice with the same matcher as the core SDK advice. IIUC, Muzzle will then ensure that it is only loaded if SQS is actually on the classpath and in a compatible version. I slightly prefer this option, but I am not sure if I should be worried about potential performance impact.
2. Reset enabled to true when the advice is applied. However, I am not sure how safe this is from a multithreading perspective without any locking/volatile (I have been burned by double checked locking despite knowing about the problem because I misunderstood something -- since then I'm quite cautious about the topic). I don't like this option very much.
3. Obviously: Keep the string equality checks on the class name as they are now, ensuring that SqsAccess is only loaded if there actually is an SQS request. However, this might be slower than native isinstance (although we don't consider subclasses). This would probably be fine as well.

Reply:
I'd add `Class.forName("software.amazon.awssdk.services.sqs.DefaultSqsClient", false, SqsAccess.class.getClassLoader())` before `SqsImpl` presence check.

Reply:
I didn't think of that simple fix, thank you! (What is the relevance of the explicit class loader? Should we also add it to the existing Class.forName call?)

That would effectively do the same as 1, wouldn't it? I now have already implemented option 1 locally and will push it for comparison. The code then looks IMHO a bit nicer since it allows to move much of the instrumentation module code into the base class. But if there are any other consequences of that which we'd like to avoid, I'm fine with the extra Class.forName for eager loading.

Reply:
I wanted to avoid initializing the class (the `false` parameter). As `DefaultSqsClient` doesn't have a static initializer it isn't too important. I didn't understand your idea before I saw the code, I think it should work too.